### PR TITLE
Add sign out dialog

### PR DESCRIPTION
**Pull request by JamieHo3312**
Show up this dialog when user click sign out button and have in progress
operations or uploads.

### CODE REVIEW

File: src/components/auth/signout/signout.html [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+      <h2 class="md-title">Would you like to sign out without your upload?</h2>
+      
+      <div class="md-body-1">
+        <p>You have in progress opreations or uploads and leaving now will cancel them. Still leaving?</p>
+      </div>
+    </md-dialog-content>
+    <md-dialog-actions>
+      <md-button class="md-primary" ng-click="@A.1"><span>Stay</span></md-button>
+      
+      <md-button class="md-primary" ng-click="@A.2"><span>Leave</span></md-button>
```
Comment (Wilson-Su) [inline src/components/auth/signout/signout.html]: You should not use the same button class here. Plz use .md-warn is better.

**Comment by JamieHo3312** [comment]
I changed the button color

**Comment by jigsawye** [comment]
I think we can use the `$mdDialog.confirm()` to implement the dialog because we don't need any input validation in the dialog, please refer the confirm dialog of [documentation](https://material.angularjs.org/latest/demo/dialog).

As before discuss, if we need any validation or other things that can't implement by javascript, then we should use your commit way to implement.

**Comment by JamieHo3312** [comment]
We can use the`$mdDialog.confirm()` to implement the dialog. But we need add `.theme()` at javascript for highlight the button, please refer the confirm dialog of [codepen](http://codepen.io/jamie3312/pen/dMzogN?editors=1010) by me.

I will push another branch for dialog style.

**Pull request closed without merging**